packages/next/src/server/app-render/walk-tree-with-flight-router-state.tsx

```diff
@@ -118,7 +117,7 @@ export async function walkTreeWithFlightRouterState({
     !experimental.isRoutePPREnabled &&
     isPrefetch &&
     !Boolean(modules.loading) &&
-    !hasLoadingComponentInTree(loaderTree)
+    !hasLoadingComponentInTree(loaderTreeToFilter)
```

Comment:
maybe for a later refactor we should extract this somewhere else. we can use whether the current modules has a loading to decide when to stop but if it is true at level 0 we really don't need to check it again at level 1 and level 2 and so on b/c it will be true until we get to the module with a loading boundary. So we can hoist it out of the recursive pathway and skip repeated checks perhaps